include/parser_interface.hpp

```diff
@@ -39,6 +39,27 @@ class ParserInterface
         return types::DbusVariantType();
     }
 
+    /**
+     * @brief API to write keyword's value on hardware.
+     *
+     * This virtual method is created to achieve runtime polymorphism for
+     * hardware writes on VPD of different types. This virtual method can be
+     * redefined at derived classess to implement write according to the type of
+     * VPD.
+     *
+     * @param[in] i_paramsToWriteData - Data required to perform write.
+     *
+     * @throw Throws exception.
+     * @return On success returns number of bytes written on hardware, On
+     * failure returns -1.
+     */
+    virtual int
+        writeKeywordOnHardware(const types::WriteVpdParams i_paramsToWriteData)
+    {
+        (void)i_paramsToWriteData;
```

Comment:
Why only here you need to do this but not in case of read? Could you please check.

Reply:
removed

Reply:
CI throws unused variable warning..
../include/parser_interface.hpp:58:60: error: unused parameter ‘i_paramsToWriteData’ [-Werror=unused-parameter]
   58 |         writeKeywordOnHardware(const types::WriteVpdParams i_paramsToWriteData)
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~